docs/decisions/0031-feature-branch-strategy.md

```diff
@@ -0,0 +1,93 @@
+
+In our current software development process, managing changes in the main branch has become increasingly complex, leading to potential conflicts and delays in release cycles.
+
+## Decision Drivers
+
+- **Pattern**: The Feature Branch Strategy is a well-known pattern for managing changes in a codebase. It is widely used in the industry and is supported by most version control systems, including GitHub, this also gives further clear picture on how the community can meaningfully contribute to the development of connectors or any other bigger feature for SK.
+
+- **Isolated Development Environments**: By using feature branches, each developer can work on different aspects of the project without interfering with others' work. This isolation reduces conflicts and ensures that the main branch remains stable.
+- **Streamlined Integration**: Feature branches simplify the process of integrating new code into the main branch. By dealing with smaller, more manageable changes, the risk of major conflicts during integration is minimized.
+- **Efficiency in Code Review**: Smaller, more focused changes in feature branches lead to quicker and more efficient code reviews. This efficiency is not just about the ease of reviewing less code at a time but also about the time saved in understanding the context and impact of the changes.
+- **Reduced Risk of Bugs**: Isolating development in feature branches reduces the likelihood of introducing bugs into the main branch. It's easier to identify and fix issues within the confined context of a single feature.
+- **Timely Feature Integration**: Small, incremental pull requests allow for quicker reviews and faster integration of features into the feature branch and make it easier to merge down into main as the code was already previously reviewed. This timeliness ensures that features are merged and ready for deployment sooner, improving the responsiveness to changes.
+
+## Considered Options
+
+- Community Feature Branch Strategy
+
+### Community Feature Branch Strategy
+
+As soon we identify that contributors are willing to take/create a Feature Issue as a potential connector implementation, we will create a new branch for that feature.
+
+The contributor(s) will then be one of the responsibles to incrementally add the majority of changes through small Pull Requests to the feature branch under our supervision and review process.
+
+This strategy involves creating a separate branch in the repository for each new big feature, like connectors. This isolation means that changes are made in a controlled environment without affecting the main branch.
+
+We may also engage in the development and changes to the feature branch when needed, the changes and full or co-authorship on the PRs will be tracked and properly referred into the Release Notes.
```

Comment:
Can you add some detail on what will need to happen before the feature branch is merged to main e.g.

1. Each connector should have a dedicated unit and integration test project
1. A unit test coverage threshold must be configured for the connector project
1.  Integration tests must be provided and the ability to run these locally and in the merge queue

Reply:
> 2\. A unit test coverage threshold must be configured for the connector project

Currently, code coverage threshold is configured for production packages only. It's not a restriction, it's just initial agreement. We can enable threshold for any package in solution at any time.
But this fact puts another question which we need to answer - what would be the rollout strategy for new connectors? Should we release it as `alpha`/`preview` first? If yes, then maybe code coverage threshold can be lower comparing to production packages. If it should be production-ready from scratch and we will release it without pre-release suffix, then code coverage threshold should be enabled for new connector in working branch before it's merged to `main`. Something that's worth to discuss, make an agreement and mention it in ADR.